test/e2e/framework/gateways.go

```diff
@@ -211,3 +211,28 @@ func gatewayNames(gateways []unstructured.Unstructured) []string {
 
 	return names
 }
+
+// Perform a gateway failover.
+// The failover for the real environment will crash the gateway node.
+// THe failover for the kind environment will set the "false" label on the gw node.
+func (f *Framework) DoFailover(cluster ClusterIndex, gwNode, gwPod string) {
+	provider := DetectProvider(cluster, gwNode)
+
+	if provider == "kind" {
+		f.SetGatewayLabelOnNode(cluster, gwNode, false)
+	} else {
+		cmd := []string{"sh", "-c", "echo 1 > /proc/sys/kernel/sysrq && echo b > /proc/sysrq-trigger"}
+
+		_, _, err := f.ExecWithOptions(&ExecOptions{
+			Command:       cmd,
+			Namespace:     TestContext.SubmarinerNamespace,
+			PodName:       gwPod,
+			ContainerName: SubmarinerGateway,
+			CaptureStdout: false,
+			CaptureStderr: true,
+		}, cluster)
+		if err != nil {
+			Errorf("Gateway failover error %q", err)
+		}
```

Comment:
It seems we should fail the test?

```suggestion
                 Expect(err).NotTo(HaveOccurred(), "Gateway failover error")
```

Reply:
@tpantelis 
Actually, there is a catch and a reason I'm not fails the test.
I'm sending the crash command to the pod and it crashes the node.
The function does not aware of which command I'm actually sending, so it tries to ensure that the command had been send properly.
But the problem is that the pod and node crashed, to it gives back an error that the pod has not been found.
So, it's like an expected error.

Maybe I can add a comment about it?
Or maybe you have some idea of how could I handle it?

Reply:
yeah at least a comment. So I assume we may or may not get an error returned depending on timing or is one always returned? It would be ideal to identify the specific error that is expected and ignore that. Is that feasible?

Reply:
I'mm add a comment.

Regarding the returned error, it always returned.
And yes, it's the same.
So you're saying that I can add a condition that if and error exists and does not contain specific error, fail the test.

Ok, will do.

Reply:
Added a condition to check specific error.
Any other error will fail the test.